libosrmc/osrmc.h

```diff
@@ -195,13 +197,17 @@ OSRMC_API float osrmc_route_response_duration(osrmc_route_response_t response, o
 
 OSRMC_API osrmc_table_params_t osrmc_table_params_construct(osrmc_error_t* error);
 OSRMC_API void osrmc_table_params_destruct(osrmc_table_params_t params);
+OSRMC_API void osrmc_table_params_set_annotations(osrmc_table_params_t params, bool return_duration, bool return_distance);
```

Comment:
Do you think we should make the annotations API open for extension from the start? That is:
- [ ] creating a new type annotations, similar to e.g. the params
- [ ] providing functions to (for now only) set the distance flag in the annotations struct
- [ ] passing the annotations struct around in the apis here

Then later on we could easily extend the annotations if users have a need for it. As it is right now it will be hard and ugly on the API to extend it later on.

Reply:
> Do you think we should make the annotations API open for extension from the start? That is:
> 
>  * [x]  creating a new type annotations, similar to e.g. the params
>  * [x]  providing functions to (for now only) set the distance flag in the annotations struct
>  * [x]  passing the annotations struct around in the apis here
> 
> Then later on we could easily extend the annotations if users have a need for it. As it is right now it will be hard and ugly on the API to extend it later on.

Sure! I've done this and defaulted it to include `AnnotationsType::Duration`, with no way to remove this. I am not sure whether OSRM even allows this!

Reply:
BTW, in doing this, I have been copying the `AnnontationsType` around, as I don't want to make the lifetime of the `osrmc_table_params_t` depend on the lifetime of the `osrmc_table_annotations_t`.